src/map/skill.cpp

```diff
@@ -18042,21 +18042,26 @@ int skill_greed(struct block_list *bl, va_list ap)
 /// Ranger's Detonator [Jobbie/3CeAM]
 int skill_detonator(struct block_list *bl, va_list ap)
 {
-	struct skill_unit *unit = NULL;
-	struct block_list *src;
-	int unit_id;
-
 	nullpo_ret(bl);
 	nullpo_ret(ap);
 
-	src = va_arg(ap,struct block_list *);
+	if (bl->type != BL_SKILL)
+		return 0;
 
-	if( bl->type != BL_SKILL || (unit = (struct skill_unit *)bl) == NULL || !unit->group )
+	block_list *src = va_arg(ap, block_list *);
+	skill_unit *unit = (skill_unit *)bl;
+
+	if (unit == nullptr)
 		return 0;
-	if( unit->group->src_id != src->id )
+
+	skill_unit_group *group = unit->group;
+
+	if (group == nullptr || group->src_id != src->id)
 		return 0;
 
-	unit_id = unit->group->unit_id;
+	int unit_id = group->unit_id;
+	uint16 skill_id = group->skill_id, skill_lv = group->skill_lv;
```

Comment:
Could be kept as group->skill_id access just like before, since all other members are still accessed via group->